### DISCUSSION TITLE
[Feature Request] Connecting Bisq to local network node

### DISCUSSION THREAD

**Discussion by SepticFrank** [raspiblitz/raspiblitz#4036] [Feature / App Ideas]
I've had trouble with bisq and tor for a long time now when connecting to my raspiblitz node. I often have to restart tor, bitcoind, or remove/add the bisq service on the raspiblitz to maintain a connection to bisq. 

One work around I've found seems to be to connect bisq to the node locally without tor, similarly to connecting to RTL or thunderhub locally. But I cannot get it to work. 

https://bisq.wiki/Connecting_to_your_own_Bitcoin_node

From the above link: If your node is on your local network, connecting directly to it rather than using Tor would reduce latency by a sensible degree. In order to do this:

    make sure the node's firewall allows incoming connections on port 8333 from the local network
    have Bitcoin daemon listen to 0.0.0.0 rather than 127.0.0.1 (add a line that says bind 0.0.0.0)
    uncheck "Use Tor for Bitcoin network" under Settings>Network in Bisq application
    fill in your node's local network ipaddress:8333 in "Use custom bitcoin Core nodes" field

I followed that exactly by editing the bitcoin.conf but I'm getting errors that electrs isn't starting and other errors. I'm wondering the best way to go about doing this without causing errors and without losing privacy. I still want to connect to bitcoin peers and lnd peers over tor, but want to connect bisq locally. How can I go about doing this?

A nice feature would be to have the option to do this with one click when setting up connecting to bisq on the raspiblitz.

**Comment by heronimoo** [comment]
I believe [this](https://github.com/raspiblitz/raspiblitz/issues/3494#issuecomment-2435019904) is also the solution here.
